csrc/scheduler/normalization_inner_outer.cpp

```diff
@@ -182,11 +182,239 @@ bool InnerOuterPersistentKernelScheduler::canScheduleCompileTime(
 
 namespace {
 
-std::pair<int64_t, int64_t> getPersistentBufferSize(
+// The roundup is due to the fact that the shared memory buffer is allocated
+// as: ceilDiv(dim_size / vectorize_factor, threads_per_block).
+// Let after_vect = dim_size / vectorize_factor;
+// n_batch = ceilDiv(after_vect, threads_per_block);
+// Then the shared memory buffer size is n_batch * vectorize_factor *
+// threads_per_block * data_type_size. This function returns the maximum
+// possible shared memory buffer size considering all possible block sizes.
+int64_t roundUpSharedMemory(
+    TensorView* tv,
```

Comment:
nitpick: not really using tv here, but rather just `int64_t data_type_size`

Reply:
revised interface.